packages/frontend/core/src/pages/auth/desktop-signin.tsx

```diff
@@ -0,0 +1,62 @@
+import { AuthService } from '@affine/core/modules/cloud';
+import { OAuthProviderType } from '@affine/graphql';
+import { useService } from '@toeverything/infra';
+import { useEffect } from 'react';
+import {
+  type LoaderFunction,
+  redirect,
+  useLoaderData,
+  // eslint-disable-next-line @typescript-eslint/no-restricted-imports
+  useNavigate,
+} from 'react-router-dom';
+import { z } from 'zod';
+
+const supportedProvider = z.nativeEnum(OAuthProviderType);
+
+interface LoaderData {
+  provider: OAuthProviderType;
+  redirectUri: string;
+}
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const url = new URL(request.url);
+  const searchParams = url.searchParams;
+  const provider = searchParams.get('provider');
+  const redirectUri = searchParams.get('redirect_uri');
+
+  // sign out first
+  await fetch('/api/auth/sign-out');
```

Comment:
Do we need to handle fetch failure branch here?

Reply:
if it fails, affine fails

Reply:
maybe should handle the network issue in the user side